micrometer-commons/build.gradle

```diff
@@ -4,6 +4,9 @@ dependencies {
     // log monitoring
     optionalApi 'ch.qos.logback:logback-classic'
 
+    // Aspects
+    optionalApi 'org.aspectj:aspectjweaver'
```

Comment:
@driessamyn Could you help us with what changes need to be made for osgi related to this. The build is failing when running the osgi-test. See, for example, https://app.circleci.com/pipelines/github/micrometer-metrics/micrometer/4025/workflows/295057ca-f327-4ed1-81dd-ba46b5f0173a/jobs/21266

Reply:
No problem, @shakuzen 

Basically, the issue can be identified here:

```
Resolution failed. Capabilities satisfying the following requirements could not be found:
    [<<INITIAL>>]
      ⇒ osgi.identity: (osgi.identity=micrometer-osgi-test-tests)
          ⇒ [micrometer-osgi-test-tests version=1.11.0.SNAPSHOT]
              ⇒ osgi.wiring.package: (&(osgi.wiring.package=io.micrometer.core.instrument.composite)(version>=1.11.0)(!(version>=2.0.0)))
                  ⇒ [micrometer-core version=1.11.0.SNAPSHOT]
                      ⇒ osgi.wiring.package: (&(osgi.wiring.package=io.micrometer.common.annotation)(version>=1.11.0)(!(version>=2.0.0)))
                          ⇒ [micrometer-commons version=1.11.0.SNAPSHOT]
                              ⇒ osgi.wiring.package: (osgi.wiring.package=org.aspectj.lang)
```

Basically, what that is saying is that `micrometer-commons` is expecting the package `org.aspectj.lang` to be "visible".
However, given this is an optional dependency, it can be marked as `dynamic` in the OSGi metadata, which means it won't try to resolve it until it's needed.

This is a patch that should resolve it: (let me know if you rather I raise a PR)

```
Index: micrometer-commons/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/micrometer-commons/build.gradle b/micrometer-commons/build.gradle
--- a/micrometer-commons/build.gradle	(revision dd86a65bb1f3d02da2bd61515d487d5fa91c0537)
+++ b/micrometer-commons/build.gradle	(date 1680507878716)
@@ -16,3 +16,14 @@
     testImplementation 'org.assertj:assertj-core'
     testImplementation 'org.awaitility:awaitility'
 }
+
+jar {
+    bundle {
+
+        bnd '''\
+        Import-Package: \
+            org.aspectj.*;resolution:=dynamic,\
+            *
+        '''.stripIndent()
+    }
+}
```

I did try to come up with a way top auto-generate the OSGi metadata based on the `optionalApi` configuration, but I couldn't think of a way to do this in gradle.

Reply:
Thanks for the help. That's fixed the build now.